code/datums/medal_awards.dm

```diff
@@ -366,3 +463,189 @@ GLOBAL_LIST_INIT(xeno_medals, list(XENO_SLAUGHTER_MEDAL, XENO_RESILIENCE_MEDAL,
 
 	return TRUE
+
+/datum/medal_recommendation
+	var/recipient_rank
+	var/recipient_ckey
+	var/recipient_name
+	var/recommended_by_name
+	var/recommended_by_ckey
+	var/reason
+	var/recommended_by_rank
+
+
+/proc/add_medal_recommendation(mob/recommendation_giver)
+	// Pick a marine
+	var/list/possible_recipients = list()
+	var/list/recipient_ranks = list()
+	for(var/datum/data/record/record in GLOB.data_core.general)
+		var/recipient_name = record.fields["name"]
+		if(recipient_name == recommendation_giver.real_name)
+			continue
+		recipient_ranks[recipient_name] = record.fields["rank"]
+		possible_recipients += recipient_name
+	var/chosen_recipient = tgui_input_list(recommendation_giver, "Who do you want to recommend a medal for?", "Medal Recommendation", possible_recipients)
+	if(!chosen_recipient)
+		return FALSE
+
+	// Write a citation
+	var/reason = strip_html(tgui_input_text(recommendation_giver, "Why does this person deserve a medal?", "Medal Recommendation", null, MAX_PAPER_MESSAGE_LEN, TRUE), MAX_PAPER_MESSAGE_LEN)
+	if(!reason)
+		return FALSE
+
+	// Get mob information
+	var/recipient_rank = recipient_ranks[chosen_recipient]
+	var/recipient_ckey
+	var/mob/recipient_mob
+	var/found_other = FALSE
+
+	for(var/mob/mob in GLOB.mob_list)
+		if(mob.real_name == chosen_recipient)
+			// We might not get this info if gibbed, so we'd need to refactor again and find another way if we want stats always correct
+			recipient_ckey = mob.persistent_ckey
+			recipient_mob = mob
+			if(found_other)
+				break
+			found_other = TRUE
+	if(!recipient_mob)
+		for(var/mob/mob in GLOB.dead_mob_list)
+			if(mob.real_name == chosen_recipient)
+				// Recipient: Check if they are dead?, and get some info
+				// We might not get this info if gibbed, so we'd need to refactor again and find another way if we want stats always correct
+				recipient_ckey = mob.persistent_ckey
+				recipient_mob = mob
+				break
+
+	// Create the recipient_award
+	var/datum/medal_recommendation/recommendation = new /datum/medal_recommendation()
+	GLOB.medal_recommendations += recommendation
+
+	recommendation.recipient_rank = recipient_rank
+	recommendation.recipient_ckey = recipient_ckey
+	recommendation.recipient_name = recipient_mob.real_name
+	recommendation.recommended_by_name = recommendation_giver.real_name
+	recommendation.recommended_by_ckey = recommendation_giver.ckey
+	recommendation.recommended_by_rank = recipient_ranks[recommendation_giver.real_name]
+
+
+	recommendation.reason = reason
+
+	return TRUE
+
+
+GLOBAL_DATUM_INIT(ic_medals_panel, /datum/ic_medal_panel, new)
+
+/datum/ic_medal_panel
+	var/name = "Medals Panel"
+	var/list/user_locs = list()
+
+/datum/ic_medal_panel/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "IcMedalsPanel", "Medals Panel")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/ic_medal_panel/ui_state(mob/user)
+	return GLOB.not_incapacitated_state
```

Comment:
At the moment you can use the tablet or computer, drop it down (in case of tablet), and run as far as way from it as you like and still interact with it.

I would suggest using a mixture of maybe not_incapacitated_and_adjacent_state and inventory_state, however as a global datum this panel has no loc (unlike the cic tablet where its ui is part of the device).

An alternative would be to implement `/datum/ic_medal_panel/ui_status/(mob/user)` but you would still need to communicate what item or machine opened this interface and test that.

So it may be best when the UI that opened this UI closes it also attempts to close any ic_medal_panel it opened.

Reply:
Fixed